include/clap/preset-features.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// This file provides a set of standard preset features meant to be used
+// within preset discovery.
+//
+// For practical reasons we'll avoid spaces and use `-` instead to facilitate
+// scripts that generate the feature array.
+//
+// Non-standard features should be formated as follow: "$namespace:$feature"
+
+#define CLAP_PRESET_FEATURE_KICK "kick"
+#define CLAP_PRESET_FEATURE_TOM "tom"
+#define CLAP_PRESET_FEATURE_SNARE "snare"
+#define CLAP_PRESET_FEATURE_CLAP "clap"
+#define CLAP_PRESET_FEATURE_CYMBAL "cymbal"
+
+#define CLAP_PRESET_FEATURE_BASS "bass"
+#define CLAP_PRESET_FEATURE_LEAD "lead"
+#define CLAP_PRESET_FEATURE_KEY "key"
+#define CLAP_PRESET_FEATURE_PLUCK "pluck"
+#define CLAP_PRESET_FEATURE_STAB "stab"
+#define CLAP_PRESET_FEATURE_WIND "wind"
+#define CLAP_PRESET_FEATURE_STRING "string"
+#define CLAP_PRESET_FEATURE_PAD "pad"
+#define CLAP_PRESET_FEATURE_DRONE "drone"
+
```

Comment:
Do we want some sort of instrument description here? I'm thinking a steinway piano model wouldn't get any of these tags for instance, nor would a Wurlitzer EP.

Reply:
(maybe the wurli would be 'key')

Reply:
Oh probably need something for tuned percussion also (marimbas etc)

Reply:
Here we try to provide a non-exhaustive list of categories and characters. So we have some kind of common language, but the plugin can use arbitrary features and the host will intelligently map it to its categorization, a bit like how google would understand your queries.

We can add more to that list.